### ISSUE TITLE
Respect default value handling

### ISSUE THREAD

**Issue by crystalgreen** [tomlm/YamlConvert#2]
Nice library!
But I have one issue: One can configure the serializer to preserve null values:

```cs
var serializer = new SerializerBuilder()
    .WithTypeConverter(new JTokenYamlConverter())
    .ConfigureDefaultValuesHandling(DefaultValuesHandling.Preserve)
    .Build();
```

However, this doesn't work - null values in JToken are always omitted. This seems to be because of this line:
[JTokenYamlConverter.cs#L199](https://github.com/tomlm/YamlConvert/blob/main/YamlConvert/JTokenYamlConverter.cs#L199).

Maybe the `JTokenYamlConverter` should always emit null and leave the default value handling to [YamlDotNet](https://github.com/aaubry/YamlDotNet/blob/master/YamlDotNet/Serialization/ObjectGraphVisitors/DefaultValuesObjectGraphVisitor.cs#L52).

**Comment by tomlm** [comment]
I have fixed this with following changes:
* I have removed the hard coded null filter as you suggest
* I have verified that null is correctly emitted or not emitted according to json.net attributes.
* I have added to JsonConvert.SerializeObject() and JsonConvert.DeserializeObject() the ability to pass in JsonSerializerSettings to control the conversion of JObject dynamic objects to YAML and back.

**Comment by tomlm** [comment]
Released as v3.0.0 https://github.com/tomlm/YamlConvert/releases/tag/3.0.0
